### ISSUE TITLE
Customize keybindings?

### ISSUE THREAD

**Issue by joh** [timbertson/slinger#7]
Is there any way to customize the keybindings? From browsing the source it looks like there should be some dconf settings in /org/gnome/shell/extensions/net/gfxmonk/slinger/keybindings, but I cannot find them when browsing in dconf-editor.

**Comment by notartom** [comment]
Yes please! Especially as at least 1 (Super-L) conflicts with the defaults (Screen lock in this case).

Edit: looking at src/menu.ts, the keybindings are hardcoded, so changing them to use dconf doesn't seem trivial (or maybe it is, but I know nothing about developing with Gnome Shell). At least it looks all localised to menu.ts.

**Comment by Pazns** [comment]
Actually the fact that Slinger steal me the Super-L in Gnome to lock the screen shortcut is completely blocker.
Very sad since this app is absolutely amazing of simplicity and efficacy.

A lot of features seem redundant in existing window managers.
Next window/Previous window ? Toggle maximize ? Maybe try to concentrate on the feature that makes you unique. This radial menu is a marvel.

**Comment by zeltak** [comment]
I love this extension! yet for me also the global keys (such as win-k) not being able to be at least disabled is really a issue. any chance at lease keys could disabled (leaving mouse support) if not possible to remap?

**Comment by jvlatzko** [comment]
> Actually the fact that Slinger steal me the Super-L in Gnome to lock the screen shortcut is completely blocker.
I simply moved Super+L to Alt+L. Worth it..

> A lot of features seem redundant in existing window managers.
> Next window/Previous window ? Toggle maximize ? Maybe try to concentrate on the feature that makes you unique. This radial menu is a marvel.

This however is a valid and good point. Next workspace (super+alt+j/k) is a straight duplicate, but seems consistent..

**Comment by DarrienG** [comment]
There are tons of conflicts with my own keybindings, which is really a shame.

**Comment by inducer** [comment]
FWIW, you can edit https://github.com/timbertson/slinger/blob/master/schemas/org.gnome.shell.extensions.net.gfxmonk.slinger.keybindings.gschema.xml, run
```
tools/gup compile
```
and install as usual. You'll need to update via git (don't forget to recompile), but that does the trick.

**Comment by timbertson** [comment]
Sorry for my silence on this. So:

 - I'm not motivated to make keybindings editable myself (because I don't need it, and I'm being selfish by only implementing what I use 😛)
 - But it's a reasonable request, and I'll happily accept a PR to implement this if anyone is willing to work on it

> Maybe try to concentrate on the feature that makes you unique. This radial menu is a marvel.

Fair point, but see above 😛  - I'm prioritising my own use, and I use these keybindings every day so if I didn't have them, the extension would be much less useful to me 🤷 

### Dconf editor:

`dconf-editor` can work, but not from a normal shell because you don't have the schemas path in `$XDG_DATA_DIRS`. 

Shellshape has some code to launch dconf-editor with the appropriate path so that this works: https://github.com/timbertson/shellshape/blob/0ffdb35524e9aa14cd12aa5dd6b5c350d46e5eea/src/gjs/shellshape_settings.ts

This is used in https://github.com/timbertson/shellshape/blob/0ffdb35524e9aa14cd12aa5dd6b5c350d46e5eea/src/gjs/prefs.ts which is the API for exposing a settings dialog in gnome-shell.

(so if anyone wants to port that, it should be possible)

### There are multiple kinds of keybindings involved:

The hardcoded ones in menu.ts are not global hotkeys, they're just the keys you use to manipulate windows while the radial menu is displayed. **I don't think anyone wants to change these** (or at least that's how I'm reading this thread, tell me if I'm mistaken).

The ones in `org.gnome.shell.extensions.net.gfxmonk.slinger.keybindings.gschema.xml` are the global hotkeys, i.e. the ones which work even when shellshape's menu isn't shown, and which can conflict with other shortcuts. These make sense to rebind, and should be reasonably simple using dconf-editor plus the code I mentioned above.
